juniper_warp/examples/warp_server/src/main.rs

```diff
@@ -0,0 +1,45 @@
+extern crate env_logger;
+#[macro_use]
+extern crate log as irrelevant_log;
+extern crate juniper;
+extern crate juniper_warp;
+extern crate warp;
+
+use juniper::tests::model::Database;
+use juniper::{EmptyMutation, RootNode};
+use warp::{http::Response, log, Filter};
+
+type Schema = RootNode<'static, Database, EmptyMutation<Database>>;
+
+fn schema() -> Schema {
+    Schema::new(Database::new(), EmptyMutation::<Database>::new())
+}
+
+fn main() {
+    ::std::env::set_var("RUST_LOG", "warp_server");
+    env_logger::init();
+
+    let log = log("warp_server");
+
+    let homepage = warp::any().map(|| {
+        Response::builder()
+            .header("content-type", "text/html")
+            .body(format!(
+                "<html><h1>juniper_warp</h1><div>visit <a href=\"/graphiql\">/graphiql</a></html>"
+            ))
+    });
+
+    info!("Listening on 127.0.0.1:8080");
+
+    let state = warp::any().map(move || Database::new());
+    let graphql_filter = juniper_warp::make_graphql_filter(schema(), state.boxed());
+
+    warp::serve(
+        warp::get(
+            warp::path("graphiql")
+                .and(juniper_warp::graphiql_handler("/graphql"))
+                .or(homepage),
+        ).or(warp::path("graphql").and(graphql_filter))
+        .with(log),
```

Comment:
with a few more days of warp experience, this doesn't look like the right way to do it, I'll have to improve this. Should this be a single file example by the way? I haven't worked with those but this one is meant to be run and used in a browser